src/main/java/io/jenkins/stapler/idea/jelly/symbols/SymbolFinder.java

```diff
@@ -0,0 +1,26 @@
+package io.jenkins.stapler.idea.jelly.symbols;
+
+import com.intellij.openapi.project.Project;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public final class SymbolFinder {
+
+    private static final ConcurrentMap<Project, Set<Symbol>> ICONS_CACHE = new ConcurrentHashMap<>();
```

Comment:
I have a hunch that holding IntelliJ project in the static field is a bad idea but I am running out of time to find the relevant documentation and IntelliJ itself doesn't seem to highlight it with a DevKit inspection 🤔

Reply:
That's fair - I've updated it to use the project name instead.